apps/prairielearn/playwright.config.ts

```diff
@@ -0,0 +1,41 @@
+import { defineConfig, devices } from '@playwright/test';
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  testDir: './src/tests/e2e',
+  fullyParallel: true,
+  forbidOnly: !!process.env.CI,
+  retries: process.env.CI ? 2 : 0,
+  workers: process.env.CI ? 1 : undefined,
+  reporter: process.env.CI
+    ? [['github'], ['blob', { outputDir: 'test-results/blob-report' }]]
+    : 'list',
+  use: {
+    baseURL: 'http://localhost:3000',
+    trace: 'on-first-retry',
+  },
+
+  projects: [
+    {
+      name: 'setup db',
+      testMatch: /global\.setup\.ts/,
```

Comment:
This approach isn't going to work. This doesn't inform the webserver of the database configuration.

Reply:
https://github.com/microsoft/playwright/issues/37377 

Unless we can have a webserver per shard, I don't see how sharding would work.

If we share a database, then we can of course wipe the database before / after the test runs. But if tests run concurrently, then we need to make sure that all the playwright tests don't interfere with each other.